src/plugins/chart_expressions/expression_xy/public/components/x_domain.tsx

```diff
@@ -42,27 +42,27 @@ export const getXDomain = (
   isHistogram: boolean
 ) => {
   const appliedTimeRange = getAppliedTimeRange(layers)?.timeRange;
+  const xValues = uniq(
+    layers
+      .flatMap<number>(({ table, xAccessor }) =>
+        table.rows.map((row) => xAccessor && row[xAccessor] && row[xAccessor].valueOf())
+      )
+      .filter(Boolean)
+      .sort()
+  );
   const from = appliedTimeRange?.from;
   const to = appliedTimeRange?.to;
   const baseDomain = isTimeViz
     ? {
-        min: from ? moment(from).valueOf() : NaN,
-        max: to ? moment(to).valueOf() : NaN,
+        min: from ? moment(from).valueOf() : xValues[0],
```

Comment:
This is breaking the endzones in a weird way in case the time field of a date histogram is not bound to the time range (in this case `appliedTimeRange` is not defined). I also think it's not "correct" because the the last data point is not the end of the chart necessarily. I'm not sure why we need it, could you elaborate? I would like to keep this logic in the elatic-charts implementation instead.

Reply:
I thought that is good fallback for time viz if we don't have appliedTimeRange.

Reply:
if there is no specific reason let's roll it back, elastic-charts should handle it already

Reply:
Also I don't understand if we don't have appliedTimeRange how the last data point will not the end of the chart?

Reply:
The min interval is also taken into account (the data point marks the beginning of a bin/bucket, not the end)

Reply:
However, passing `NaN` here is the official way of telling elastic-charts there is no defined min/max which is exactly the case we are dealing with here.

Reply:
esdocs don't provide not only appliedTimeRange but also interval.

Reply:
> However, passing `NaN` here is the official way of telling elastic-charts there is no defined min/max which is exactly the case we are dealing with here.

Okay, I will check if annotations will work with that way.

Reply:
@flash1293 Annotations work fine with NaN as min/max for domain, I'll remove my fallback